subprojects/docs/src/docs/userguide/console.adoc

```diff
@@ -0,0 +1,145 @@
+
+NOTE: Grouped console output and reduced console output only occurs with interactive and _rich console_ command-lines. Continuous integration servers and builds using `--console=plain` will see console output similar to pre-Gradle 4.0.
+
+The following console output shows grouped output for the configuration phase and the task `:compileJava`:
+
+----
+> Configure project ':library'
+Configuring project version for project ':library'
+
+> Configure project ':consumer'
+Configuring project version for project ':consumer'
+
+> Task :compileJava
+Note: Some input files use unchecked or unsafe operations.
+Note: Recompile with -Xlint:unchecked for details.
+----
+
+Gradle does not wait until a unit of work is fully completed before displaying its output. Gradle flushes output to the console after a short amount of time to ensure that relevant information is made available as soon as possible. When building in parallel, the output of long running tasks can be broken up by other tasks. Each block of console output will clearly indicate which task it belongs to.
+
+----
+> Task :compileJava
+Note: Some input files use unchecked or unsafe operations.
+Note: Recompile with -Xlint:unchecked for details.
+
+> Task :generateCode
+Generating JAXB classed from XSD files.
+
+> Task :compileJava
+Note: Some input files use or override a deprecated API.
+Note: Recompile with -Xlint:deprecation for details.
+----
+
+[[sec:console_build_progress_bar]]
+==== Build progress bar
+
+The build progress bar gives you a very fast way of knowing if the build will be finished soon. As the build performs work, the progress bar will fill from left to right. At any given time, the build progress bar also renders the current phase of the build lifecycle (see <<sec:build_phases>>) and the overall time spent during the build.
+
+The following examples shows the progress bar during the initialization, configuration and execution phase of the build lifecycle:
+
+----
+<-------------> 0% INITIALIZING [2s]
+<==-----------> 25% CONFIGURING [4s]
+<=========----> 64% EXECUTING [17s]
+----
+
+[[sec:console_work_in_progress_display]]
+==== Work in-progress display
+
+Gradle provides a fined-grained view of the actual work being performed directly underneath the <<sec:console_build_progress_bar,The build progress bar>>. Each line represents a thread or process that can perform work in parallel--resolving dependencies, executing a task and running tests. If an available worker is not being used then it is marked with `IDLE`. The number of available workers defaults to the number of processors on the machine executing the build. You can adjust the number of workers through the <<gradle_command_line,command line>> option `--max-workers`.
```

Comment:
:x: we should note that these lines do not map exactly to workers and that some background process that never submits events will never show up.

We should also note that this does not yet display parallel test execution.